### DISCUSSION TITLE
Is it right that I can only set a language to  EXTRA_ARGS='--language {lang}'  ?

### DISCUSSION THREAD

**Discussion by Innocence** [compiler-explorer/compiler-explorer#7074] [Q&A] [answered]
Or is there a way to run compiler-explorer  within specific languages?

I set up compiler-explorer in Windows locally, when I run **npm start** ,  http://localhost:10240/   included all languages. 
When I run **make dev EXTRA_ARGS='--language c++ python'**, only C++  showed.
I tried edit **languages.ts** and **languages.interfaces.ts**, but anything changed to those two files , compiler-explorer won't start.

Sorry, I know nothing about typescript &  front-end . Need helps about this.

**Comment by Innocence** [comment]
I  tried those commands .

make dev EXTRA_ARGS='--language c++ python' 
make dev EXTRA_ARGS='--language 'c++ python''
make dev EXTRA_ARGS='--language [c++ python]'
make dev EXTRA_ARGS='--language {c++ python}'

**Comment by junlarsen** [comment]
We use https://www.npmjs.com/package/nopt argument parsing. I believe you should be able to do the following: `--language c++ --language python`

**Reply by Innocence** [reply]
> We use https://www.npmjs.com/package/nopt argument parsing. I believe you should be able to do the following: `--language c++ --language python`

Tried these 3 ways, only last language worked.

make dev EXTRA_ARGS='--language c++ --language python'   
make dev EXTRA_ARGS='--language c++'  EXTRA_ARGS='--language python'
make dev EXTRA_ARGS='--language c++'  '--language python'

**Comment by mattgodbolt** [comment]
Well...something has broken on our side! I'm sure this used to work but I just tested and it doesn't! Taking a look now...

**Comment by mattgodbolt** [comment]
No - this looks like it never worked...oops :) Fixing now...

**Comment by mattgodbolt** [comment, marked as answer]
Ok! Looks like this needs a comma-separated list of languages:

```
make dev EXTRA_ARGS="--language c++,rust"
```

just worked for me. I'll make it work for both though!
